File: mobile_sejourfr/lib/core/models/progression_status.dart
/// **Le seul vocabulaire d'état pédagogique du mobile** — miroir de
/// `ProgressionStatus` (backend, moteur de progression V4.2 §13, §25 bis.3).
///
/// 🛑 **Le front ne calcule aucun état, aucun niveau CECRL, aucun seuil**
/// (invariant I42). Cet enum se lit dans un DTO servi, jamais d'un pourcentage :
/// le libellé et le ton arrivent avec lui, ils ne se déduisent pas ici. Les
/// valeurs de repli présentes ci-dessous ne sont là que pour un rendu offline
/// cohérent — elles recopient le contrat, elles ne l'inventent pas.
///
/// Les six états doivent tous être rendus. Une UI qui n'en gère que quatre est
/// non conforme : `watch` et `readyForReassessment` sont précisément ceux qui
/// portent la valeur pédagogique du produit, et ce sont les premiers qu'on perd
/// en recopiant un ancien vocabulaire.
library;

enum ProgressionStatus {
  /// Jamais mesuré directement. **Pas de pourcentage affiché** : un palier sans
  /// preuve directe vaut `null`, jamais 0 % (§18.6, invariant I41).
  notEvaluated('NOT_EVALUATED', 'À évaluer', ProgressionTone.neutral),

  /// Des preuves existent, sous le seuil d'entrée en progression.
  fragile('FRAGILE', 'À renforcer', ProgressionTone.danger),

  /// Score et confiance suffisants pour progresser.
  progressing('PROGRESSING', 'En progression', ProgressionTone.primary),

  /// Prêt à être vérifié sur une vraie tâche — EE/EO uniquement.
  readyForReassessment(
      'READY_FOR_REASSESSMENT', 'Prêt à vérifier', ProgressionTone.accent),

  /// Palier ou compétence confirmé.
  solid('SOLID', 'Acquis', ProgressionTone.success),

  /// Un acquis contredit une fois : vérification ciblée prioritaire.
  watch('WATCH', 'À vérifier', ProgressionTone.warn);

  const ProgressionStatus(this.wire, this.fallbackLabel, this.tone);

  final String wire;

  /// Le libellé **servi** fait foi ; celui-ci ne sert qu'à défaut.
  final String fallbackLabel;

  final ProgressionTone tone;

  static ProgressionStatus fromWire(String value) =>
      ProgressionStatus.values.firstWhere(
        (status) => status.wire == value,
        orElse: () => ProgressionStatus.notEvaluated,
      );
}

/// Le ton visuel d'un état — **dérivé de l'état, jamais d'un nombre**.
///
/// Rappel charte : le rouge est réservé aux CTA critiques et aux signaux
/// d'urgence. Seul `danger` l'utilise ; les autres prennent le bleu, le vert ou
/// l'ambre.
enum ProgressionTone { neutral, danger, primary, accent, success, warn }

File: backend_sejourfr/src/main/java/com/sejourfr/app/progression/domain/ProgressionStatus.java
package com.sejourfr.app.progression.domain;

/**
 * Les six etats normalises de la progression (V4.2 §13) et leur libelle FR.
 *
 * <p><b>Le libelle est servi par le serveur</b> (§25 bis.2) : aucun front ne
 * traduit cet enum et aucun front ne le derive d'un nombre. Une UI qui ne
 * gererait que quatre etats est non conforme — {@link #WATCH} et
 * {@link #READY_FOR_REASSESSMENT} sont precisement ceux qui portent la valeur
 * pedagogique du produit (§25 bis.3).
 *
 * <p>Le {@code tone} visuel se derive de cet etat, <b>jamais d'un
 * pourcentage</b> (invariant I42).
 */
public enum ProgressionStatus {

    /** Aucune preuve directe : {@code sumWeightEpoch == 0} (§14.1). */
    NOT_EVALUATED("À évaluer", "neutral"),

    /** Des preuves existent, sous le seuil d'entree en progression (§14.2). */
    FRAGILE("À renforcer", "danger"),

    /** Score et confiance suffisants pour progresser (§14.2). */
    PROGRESSING("En progression", "primary"),

    /** EE/EO uniquement : pret a etre verifie sur une vraie tache (§14.3). */
    READY_FOR_REASSESSMENT("Prêt à vérifier", "accent"),

    /** Palier ou competence confirme, gate satisfait (§14.4, §14.5). */
    SOLID("Acquis", "success"),

    /** Un SOLID contredit une fois : verification ciblee prioritaire (§15). */
    WATCH("À vérifier", "warn");

    private final String label;
    private final String tone;

    ProgressionStatus(String label, String tone) {
        this.label = label;
        this.tone = tone;
    }

    /** Libelle FR servi au front — contrat gele (§25 bis.3). */
    public String getLabel() {
        return label;
    }

    /** Ton visuel servi au front, derive de l'etat et jamais d'un nombre. */
    public String getTone() {
        return tone;
    }
}
